examples_gbs/run_tutorial_similarity.py

```diff
@@ -0,0 +1,278 @@
+# samples :math:`\{S_{1}, S_{2}, \ldots , S_{N}\}` are generated, then the event probability can
+# be approximated as
+#
+# .. math::
+#     p(E_{k, n_{\max}}) \approx \frac{1}{N}\sum_{i=1}^N p(S_i) |E_{k, n_{\max}}|,
+#
+# with :math:`|E_{k, n_{\max}}|` denoting the cardinality of the event.
+#
+# This method can be accessed using the :func:`~.prob_event_mc` function. The 4-photon event is
+# approximated as:
+
+print(similarity.prob_event_mc(nx.Graph(m0_a), 4, max_count_per_mode=2, n_mean=6))
+
+##############################################################################
+# The feature vector can then be calculated through Monte Carlo sampling using
+# :func:`~.feature_vector_mc`.
+#
+# .. note::
+#     The results of :func:`~.prob_event_mc` and :func:`~.feature_vector_mc` are probabilistic and
+#     may vary between runs. Increasing the optional ``samples`` parameter will increase accuracy
+#     but slow down calculation.
+#
+# The second method of MC approximation is intended for use in scenarios where it is
+# computationally intensive to pre-calculate a statistically significant dataset of samples from
+# GBS.
+#
+# Machine learning with GBS graph kernels
+# ---------------------------------------
+#
+# We have seen how GBS can be used to provide a mapping of graphs into a feature space. This
+# mapping can be used for machine learning tasks such as classification: by viewing each graph as
+# a point in the high dimensional space, we can use standard approaches from machine learning
+# such as `support vector machines <https://en.wikipedia.org/wiki/Support-vector_machine>`__ (SVMs).
+#
+# Let's build this up a bit more. The MUTAG dataset we are considering contains not only graphs
+# corresponding to the structure of chemical compounds, but also a *classification* of each
+# compound based upon its mutagenic effect. The four graphs we consider here have classifications:
+#
+# - MUTAG0: Class 1
+# - MUTAG1: Class 0
+# - MUTAG2: Class 0
+# - MUTAG3: Class 1
+
+classes = [1, 0, 0, 1]
+
+##############################################################################
+# Can we use GBS feature vectors to classify these graphs? We start by defining two-dimensional
+# feature vectors:
+
+events = [8, 10]
+max_count = 2
+
+f1 = similarity.feature_vector_sampling(m0, events, max_count)
+f2 = similarity.feature_vector_sampling(m1, events, max_count)
+f3 = similarity.feature_vector_sampling(m2, events, max_count)
+f4 = similarity.feature_vector_sampling(m3, events, max_count)
+
+import numpy as np
+
+R = np.array([f1, f2, f3, f4])
+
+print(R)
+
+##############################################################################
+# Given our points in the feature space and corresponding classifications, we can use scikit learn
+# to train an SVM:
+
+from sklearn.svm import LinearSVC
+from sklearn.preprocessing import StandardScaler
+
+R_scaled = StandardScaler().fit_transform(R)  # Transform data to zero mean and unit variance

```

Comment:
Why do we transform data this way?

Reply:
Honest reason:
The fit looked weird without doing it, and seemed to work better with scaling.

Nicer sounding reason:
I believe it's actually a fairly standard thing to do:
https://scikit-learn.org/stable/modules/preprocessing.html

> Standardization of datasets is a common requirement for many machine learning estimators implemented in scikit-learn; they might behave badly if the individual features do not more or less look like standard normally distributed data: Gaussian with zero mean and unit variance.

Do you think this should be explained in the text?

Reply:
Yes, I think it's best to explain. It can be a single short sentence, maybe linking to the URL you posted here

Reply:
Great, explained! Thanks